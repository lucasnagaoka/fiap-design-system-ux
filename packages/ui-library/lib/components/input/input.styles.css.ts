import { style } from '@vanilla-extract/css';

import { FIAP_DS_COLORS } from '../../design-tokens/colors';
import { FIAP_DS_SPACING } from '../../design-tokens/spacing';
import { FIAP_DS_TYPOGRAPHY } from '../../design-tokens/typography';

const input = style({
  padding: FIAP_DS_SPACING.s2,
  borderRadius: FIAP_DS_SPACING.s2,
  border: 0,
  width: '100%',
  fontSize: FIAP_DS_TYPOGRAPHY.button['font-size'],
  lineHeight: FIAP_DS_TYPOGRAPHY.button['line-height'],
  fontWeight: FIAP_DS_TYPOGRAPHY.button.weight,

  ':focus-visible': {
    outline: `1px solid ${FIAP_DS_COLORS.white[100]}`,
  },
});

const inputPrimary = style({
  backgroundColor: FIAP_DS_COLORS.primary[900],
  color: FIAP_DS_COLORS.white[100],

  '::placeholder': {
    color: FIAP_DS_COLORS.white[100],
  },
});

const inputSecondary = style({
  backgroundColor: FIAP_DS_COLORS.white[100],
  color: FIAP_DS_COLORS.primary[700],
  border: `1px solid ${FIAP_DS_COLORS.primary[700]}`,

  '::placeholder': {
    color: FIAP_DS_COLORS.primary[700],
  },
});

export { input, inputPrimary, inputSecondary };
