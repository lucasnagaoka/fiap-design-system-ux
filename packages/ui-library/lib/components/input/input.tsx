import clsx from 'clsx';
import { fontText } from '../../styles/font.css';
import { input, inputPrimary, inputSecondary } from './input.styles.css';

import type { JSX } from 'react';
import type { InputProps, InputVariant } from './input.types';

export function Input({
  variant = 'primary',
  ...other
}: InputProps): JSX.Element {
  function getVariantClass(variant: InputVariant) {
    switch (variant) {
      case 'primary':
        return inputPrimary;
      case 'secondary':
        return inputSecondary;
      default:
        return input;
    }
  }

  return (
    <input
      className={clsx([fontText, input, getVariantClass(variant)])}
      {...other}
    />
  );
}

Input.displayName = 'Input';
