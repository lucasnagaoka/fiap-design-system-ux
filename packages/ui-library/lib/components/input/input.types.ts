export type InputVariant = 'primary' | 'secondary';

export type InputProps = React.InputHTMLAttributes<HTMLInputElement> & {
  variant?: InputVariant;
};
