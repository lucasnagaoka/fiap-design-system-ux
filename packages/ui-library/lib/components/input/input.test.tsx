import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Input } from './input';
import * as styles from './input.styles.css.ts';

describe('Input component', () => {
  it('Should render a primary input', () => {
    render(<Input variant='primary' placeholder='meu input' />);
    const input = screen.getByPlaceholderText('meu input');

    expect(input.classList.contains(styles.input)).toBeTruthy();
    expect(input.classList.contains(styles.inputPrimary)).toBeTruthy();
  });

  it('Should render a secondary input', () => {
    render(<Input variant='secondary' placeholder='meu input' />);
    const input = screen.getByPlaceholderText('meu input');

    expect(input.classList.contains(styles.input)).toBeTruthy();
    expect(input.classList.contains(styles.inputSecondary)).toBeTruthy();
  });

  it('Should render the primary input if no variant is passed.', () => {
    render(<Input placeholder='meu input' />);
    const input = screen.getByPlaceholderText('meu input');

    expect(input.classList.contains(styles.input)).toBeTruthy();
    expect(input.classList.contains(styles.inputPrimary)).toBeTruthy();
    expect(input.classList.contains(styles.inputSecondary)).toBeFalsy();
  });

  it('Should take a input snapshot', () => {
    const { asFragment } = render(
      <Input placeholder='Email' variant='primary' />
    );

    expect(asFragment).toMatchSnapshot();
  });
});
