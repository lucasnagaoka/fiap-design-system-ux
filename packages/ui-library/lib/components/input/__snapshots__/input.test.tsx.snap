// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input component > Should take a input snapshot 1`] = `[Function]`;
